### PR TITLE
API additions and socket stuff

### PR DESCRIPTION
**Pull request by reza-hashemi**
Added create/start apis
Added players via socket connections

### CODE REVIEW

File: src/server/src/classes/Game.ts [hunk 1]
```diff
@@ -11,8 +11,9 @@ export default class Game {
   private gameState: IGame;
   private currentRound: Round;
 
-  constructor() {
+  constructor(id: string) {
```
[the remark this answers is not in the document]
Comment (austinberke) [reply src/server/src/classes/Game.ts]: doing it this way makes it easier to add/change params in the future

File: src/server/src/simulatePlayer.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import io from 'socket.io-client';
```
Comment (austinberke) [inline src/server/src/simulatePlayer.ts]: consider moving simulation stuff into a `simulate` folder

File: src/server/src/tests/deck.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { expect } from 'chai';
```
Comment (austinberke) [inline src/server/src/tests/deck.ts]: this shouldn't show as part of this PR. try running a `git pull origin master`

File: src/server/src/app.ts [hunk 4]
```diff
@@ -9,35 +17,80 @@ import { v1 as uuid } from 'uuid';
...
+app.get('/game/:id', (req, res) => {
+  const { id } = req.params;
+  if (games.has(id)) {
+    // TODO: send game front end code
+    res.send({
+      id,
+    });
   } else {
-    res.send('game does not exist, sorry :(');
+    res.status(4040).send('game does not exist, sorry :(');
```
Comment (austinberke) [inline src/server/src/app.ts]: 404

File: src/server/src/tests/deck.ts (hunk 3 above)
Comment (reza-hashemi) [reply src/server/src/tests/deck.ts]: Ran it but it says that im up to date. I must have merged incorrectly somehow.

Comment (austinberke) [reply src/server/src/tests/deck.ts]: oh wait this shouldn't be here. this should be called `deck.test.ts`

**Pull request merged**